GomokuClient/packages/gomoku-core/src/hooks/useBoard.ts

```diff
@@ -1,12 +1,18 @@
 import { useCallback, useEffect, useRef, useState } from "react";
 import { findWinner, Winner } from "@/utils";
 
-export type CellValue = "black" | "white" | null;
+export type CellValue = "Black" | "White" | null;
 
+// TODO: This hook probably should be refactored.
+// Ideally it should be responsible just for keeping board state with possible add new pieces
+// And as well it should accept boardInitialState as incoming parameter to be able to recover game
+// when active game state received from server
+// Every other logic, like calculating winner and keeping in state what player color made move should be
+// removed and received from server
```

Comment:
Let me challenge the suggestions mentioned in this TODO:


- While it’s true that some of the logic should be moved to the server to maintain consistency across multiple clients, calculating the winner locally and determining the next player makes sense from a UI perspective. This approach allows for immediate feedback, improving the user experience, while the server can still validate game integrity in parallel.

- I plan to create two hooks: one called useLocalBoard for local gameplay on a single device, and another that better aligns with server requests and accepts relevant props. Also, considering we will have an AI player, the current approach fits well in terms of simplicity.

- The current board representation (Array(19).fill(...)) is straightforward and easy to understand, especially since it’s just a matrix representation. While using a Record<string, MoveData> could optimize access to specific cells, it might add unnecessary complexity. We need to evaluate if the performance gains are worth it.

- Regarding the use of useCallback and memoization, I’m not focusing on that for now, especially with React 19’s upcoming automatic memoization features. In general, I intend to integrate an ESLint plugin that will automatically highlight when memoization is required and flag missing dependencies. As you pointed out, the useCallback here is indeed unnecessary and can be removed. However, I recommend integrating the ESLint plugin first. Here’s the reference: [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks).

Reply:
- From UI perspective probably it's better, but what if request does not reaches server and last move state is not saved? So, you show, that game is over, while at server it still ongoing. But that edge case, of course, and not worth to dig into scenarios like that now

- Okay, sounds good

- I don't think it's somehow complex. You will just iterate in for loop and get values by keys built from for loop indexes. And will get value by key, which will be "black" | "white" | undefined. This is much better if you need to access square information by indexes. But that just thoughts for the future. As we talked, maybe we will introduce another formats

- Okay, got it. Let's keep it for another PR